src/mono/mono/mini/interp/jiterpreter.c

```diff
@@ -942,6 +942,30 @@ trace_info_alloc () {
 	return index;
 }
 
+static void
+build_address_taken_bitset (TransformData *td, InterpBasicBlock *bb, guint32 bitset_size)
```

Comment:
If you want, you can further optimize this a bit. For example, you can add a new field to `TransformData`, something like `max_indirect_offset` that you set in https://github.com/dotnet/runtime/blob/d0913fce8975849f90fd232698d16c3f95d89d6d/src/mono/mono/mini/interp/transform.c#L10316 to be `td->total_locals_size`. You can then run this pass only if this field is non zero and you can also have less memory use for the bitset array (note you would need to do a range check in jiterp for overflowing offsets)

Reply:
That sounds great, but I'd prefer to do it in a follow-up PR. Is that OK?

Reply:
sure, also if you consider this to be worth it